rules/table-of-contents.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+const visit = require('unist-util-visit');
+const chalk = require('chalk');
+
+module.exports = (ast, file, language, done) => {
+	visit(ast, 'heading', (node, index) => {
+		const title = node.children[0].value;
+
+		if (index === 1 && title !== 'Contents') {
```

Comment:
`unist` doesn’t parse anything, it’s a schema of describing syntaxes, so, `{type: 'heading', depth: 1}` is an [MDAST](https://github.com/wooorm/mdast) node (MDAST implements Unist)!

remark does parse markdown, but it doesn’t parse HTML. To markdown, `<h1>foo</h1>` is just a string of characters, without meaning. It’s not a heading in markdown, even though it’s rendered on GitHub as an HTML heading. But, to this rule that doesn’t matter. I don’t think there’s a reason to use `<h2>Contents</h2>`, so filtering all top-level headings, removing the first if it’s meant as the main heading, and then checking the first, should do the trick!

The reason to check `ast.children`, instead visiting the whole ast, is because the former also walks into blockquotes and lists, which also may have headings. As we want the `Contents` heading to match top-level only, there’s no need to walk the complete tree (as its slower, and could give false positives.)